futures-util/src/sink/mod.rs

```diff
@@ -223,7 +223,7 @@ pub trait SinkExt<Item>: Sink<Item> {
         &'a mut self,
         stream: &'a mut St
     ) -> SendAll<'a, Self, St>
-        where St: Stream<Item = Item> + Unpin,
+        where &'a mut St: TryStream<Ok = Item, Error = Self::Error> + Unpin,
```

Comment:
I was thinking about using `TryStream + Stream` bounds like [Stream::forward](https://github.com/rust-lang-nursery/futures-rs/blob/0c165a30954c98f281311ae7ce9be26491ca2932/futures-util/src/stream/stream/forward.rs#L25) - but I don't have any strong opinion on this.